docs/source/app-dev/services.rst

```diff
@@ -68,20 +73,64 @@ For full details, see :ref:`the proto documentation for the service <com.daml.le
 Command deduplication
 ---------------------
 
-The command submission service deduplicates submitted commands based on the submitting :ref:`party <com.daml.ledger.api.v1.Commands.party>` and :ref:`command ID <com.daml.ledger.api.v1.Commands.command_id>`:
+The command submission service deduplicates command submissions based on the change ID.
+The change ID consists of the following:
+
+* The :ref:`application ID <com.daml.ledger.api.v1.Commands.application_id>`
+* The :ref:`command ID <com.daml.ledger.api.v1.Commands.command_id>`
+* The set of submitting parties, i.e., the union of :ref:`party <com.daml.ledger.api.v1.Commands.party>` and :ref:`actAs <com.daml.ledger.api.v1.Commands.actAs>`
+
+Command deduplication obeys the following rules:
+
+#. Applications can specify the start of the :ref:`deduplication period <com.daml.ledger.api.v1.Commands.deduplication>` for each command submission.
+   If this parameter is not set, the participant chooses the start at its discretion.
+   
+#. **Deduplication guarantee:**
+   A command submission is considered a duplicate submission if another command with the same change ID is already in flight or there is an accepting command completion since the start of the deduplication period with the same change ID.
+   Ledgers are free to extend the deduplication period beyond the start of the deduplication period.
+   Commands with rejected completions do not trigger deduplication.
+
+#. A duplicate command submission leads to an error as an RPC response or in a completion with the status codes ``ABORTED`` for in-flight conflicts and ``ALREADY_EXISTS`` for already accepted commands.
+   The error details contain information about the conflicting command submission.
```

Comment:
Do we have a place where we explain the error reporting approach employed by the Ledger API? Should we refer to the docs generated from the gRPC definition of `error.proto`?

Reply:
I guess that we will flesh out `error.proto` into a proper `.rst` file and add it to the documentation webpage.